--- nmh-1.0.3/config.h.in.config	Mon Feb 28 14:17:24 2000
+++ nmh-1.0.3/config.h.in	Mon Feb 28 14:18:56 2000
@@ -23,8 +23,8 @@
  * other programs which may modify your maildrops.
  * Currently you can only use one type.
  */
-#define DOT_LOCKING   1
-/* #define FCNTL_LOCKING 1 */
+/* #define DOT_LOCKING   1 */
+#define FCNTL_LOCKING 1
 /* #define LOCKF_LOCKING 1 */
 /* #define FLOCK_LOCKING 1 */
 
@@ -93,7 +93,7 @@
  * to be able to use both POP3 _and_ Kerberized POP and plan to use inc and
  * msgchk's -kpop switch every time in the latter case.
  */
-#define POPSERVICE "pop3"
+#define POPSERVICE "pop-3"
 
 /*
  * Define the default creation modes for folders and messages.
@@ -139,7 +139,7 @@
 #undef SCO_5_STDIO
 
 /* This tells nmh to use the Linux _IO_*_ptr defines from libio.h */ 
-#undef LINUX_STDIO
+#define LINUX_STDIO 1
 
 
 /***** END USER CONFIGURATION SECTION *****/
