--- nmh-1.2.orig/etc/Makefile.in	2006-01-12 11:19:12.000000000 +0100
+++ nmh-1.2/etc/Makefile.in	2006-01-12 12:12:14.000000000 +0100
@@ -72,13 +72,13 @@
 	rm -f $@
 	$(SED) -e 's,%mts%,$(MTS),' \
 	       -e 's,%mailspool%,$(mailspool),' \
-	       -e 's,%etcdir%,$(etcdir),' \
+	       -e 's,%etcdir%,$(DESTDIR)$(etcdir),' \
 	       -e 's,%masquerade%,$(masquerade),' \
 	       -e 's,%smtpservers%,$(smtpservers),' < $(srcdir)/mts.conf.in > $@
 
 sendfiles: $(srcdir)/sendfiles.in Makefile
 	rm -f $@
-	$(SED) -e 's,%libdir%,$(libdir),' < $(srcdir)/sendfiles.in > $@
+	$(SED) -e 's,%libdir%,$(DESTDIR)$(libdir),' < $(srcdir)/sendfiles.in > $@
 
 install: install-bin-files install-etc-files
 
--- nmh-1.2.orig/man/Makefile.in	2006-01-12 11:19:12.000000000 +0100
+++ nmh-1.2/man/Makefile.in	2006-01-12 12:14:33.000000000 +0100
@@ -50,7 +50,7 @@
 .man.$(manext8):
 	$(SEDMAN)
 
-# man pages to install in $(mandir)/$(manext1)
+# man pages to install in $(DESTDIR)/$(mandir)/$(manext1)
 MAN1SRC = ali. anno. burst. comp.		\
        dist. flist. flists. folder. folders.	\
        forw. inc. install-mh. mark. mh-chart.	\
@@ -99,10 +99,10 @@
 	echo 's,%nmhwarning%,THIS FILE HAS BEEN AUTOMATICALLY GENERATED.  DO NOT EDIT.,g' > $@
 	echo 's,%nmhversion%,nmh-$(VERSION),g' >> $@
 	echo 's,%nmhdate%,$(DATE),g' >> $@
-	echo 's,%bindir%,$(bindir),g' >> $@
+	echo 's,%bindir%,$(DESTDIR)$(bindir),g' >> $@
 	echo 's,%etcdir%,$(etcdir),g' >> $@
-	echo 's,%libdir%,$(libdir),g' >> $@
-	echo 's,%mandir%,$(mandir),g' >> $@
+	echo 's,%libdir%,$(DESTDIR)$(libdir),g' >> $@
+	echo 's,%mandir%,$(DESTDIR)$(mandir),g' >> $@
 	echo 's,%mailspool%,$(mailspool),g' >> $@
 	echo 's,%sendmailpath%,$(sendmailpath),g' >> $@
 	echo 's,%default_editor%,$(default_editor),g' >> $@
